functional_tests.go

```diff
@@ -2166,14 +2166,32 @@ func testFunctional() {
 		logger().Fatal("Error: ", err)
 	}
 
+	// Generate presigned HEAD object url.
+	presignedHeadURL, err := c.PresignedHeadObject(bucketName, objectName, 3600*time.Second, nil)
```

Comment:
Can you please use another instance of Minio client initialized without credentials ? We will be more close to real world use case